backend/diagnostics.go

```diff
@@ -83,3 +90,96 @@ func (fn CollectMetricsHandlerFunc) CollectMetrics(ctx context.Context) (*Collec
 type CollectMetricsResult struct {
 	PrometheusMetrics []byte
 }
+
+// ToUsageStats filters a CollectMetricsResult and returns usage stats
+func (res *CollectMetricsResult) ToUsageStats(pluginID string) map[string]interface{} {
+	var (
+		parser expfmt.TextParser
+		r      = regexp.MustCompile(`usage_stats[_]?`)
+		stats  = map[string]interface{}{}
+		reader = bytes.NewReader(res.PrometheusMetrics)
+	)
+
+	mfs, err := parser.TextToMetricFamilies(reader)
+	if err != nil {
+		Logger.Error("unable to parse usage stats from metric", "plugin", pluginID, "err", err)
+	}
+	for _, mf := range mfs {
+		if !strings.HasPrefix(mf.GetName(), "usage_stats") {
+			continue
+		}
+		name := r.ReplaceAllString(mf.GetName(), "")
+		for _, m := range mf.Metric {
+			labels := []string{pluginID, name}
+			for _, l := range m.Label {
+				labels = append(labels, l.GetName(), l.GetValue())
+			}
+			converted := map[string]interface{}{}
+			switch mf.GetType() {
+			case prom.MetricType_COUNTER:
+				converted = convertCounter(labels, m)
+			case prom.MetricType_GAUGE:
+				converted = convertGauge(labels, m)
+			case prom.MetricType_SUMMARY:
+				converted = convertSummary(labels, m)
+			case prom.MetricType_HISTOGRAM:
+				converted = convertHistogram(labels, m)
+			}
+			for k, v := range converted {
+				stats[k] = v
+			}
+		}
+	}
+
+	return stats
+}
+
+func convertCounter(labels []string, m *prom.Metric) map[string]interface{} {
+	stats := map[string]interface{}{}
+	key := toUsageStatKey(labels...)
+	stats[key] = m.GetCounter().GetValue()
```

Comment:
Not sure I understand how counters are helpful in the context of usage stats, when we only get a single reading each day. I guess if the counters are slow enough that they take more than 24hrs to wrap we could look at the value from the previous report to try and come up with a rate.  Ideally we'd want to just report a rate directly to avoid needing to do that server-side.